src/coreclr/jit/emit.cpp

```diff
@@ -6773,12 +6773,10 @@ unsigned emitter::emitEndCodeGen(Compiler*         comp,
 #endif
 
 #if defined(TARGET_XARCH) || defined(TARGET_ARM64)
-    // For x64/x86/arm64, align methods that are "optimizations enabled" to 32 byte boundaries if
-    // they are larger than 16 bytes and contain a loop.
+    // For x64/x86/arm64, align methods that contain a loop to 32 byte boundaries if
+    // they are larger than 16 bytes and loop alignment is enabled.
     //
-    if (emitComp->opts.OptimizationEnabled() &&
-        (!emitComp->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_PREJIT) || comp->IsTargetAbi(CORINFO_NATIVEAOT_ABI)) &&
-        (emitTotalHotCodeSize > 16) && emitComp->fgHasLoops)
+    if (codeGen->ShouldAlignLoops() && (emitTotalHotCodeSize > 16) && emitComp->fgHasLoops)
```

Comment:
Should we also set `JitAlignLoops` to `false` if we are optimizing for size and hence, here we will also adjust the method alignment to 8 bytes instead of 32 bytes?

Reply:
I’d say yes. I wanted to unblock my testing first which is why I submitted this separately.